### PR TITLE
fix(obstacle_stop_planner): lock object_ptr_, current_velocity_ptr_

### PR DESCRIPTION
**Pull request by satoshi-ota**
Signed-off-by: satoshi-ota <satoshi.ota928@gmail.com>

## Description

- lock member variables (obejct_ptr_, current_velocity_ptr_)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/obstacle_stop_planner/src/node.cpp [hunk 1]
```diff
@@ -990,13 +996,17 @@ SlowDownSection ObstacleStopPlannerNode::createSlowDownSection(
...
   const VehicleInfo & vehicle_info, const double current_acc)
 {
+  mutex_.lock();
   if (!current_velocity_ptr_) {
     // TODO(Satoshi Ota)
+    mutex_.unlock();
     return SlowDownSection{};
   }
 
+  const auto & current_vel = current_velocity_ptr_->twist.twist.linear.x;
```
Comment (yukkysaito) [inline planning/obstacle_stop_planner/src/node.cpp:1006]: It maybe changed `current_velocity_ptr_->twist.twist.linear.x` after unlock
```suggestion
  const auto current_vel = current_velocity_ptr_->twist.twist.linear.x;
```

File: planning/obstacle_stop_planner/src/node.cpp [hunk 2]
```diff
@@ -1518,7 +1533,10 @@ void ObstacleStopPlannerNode::resetExternalVelocityLimit(const double current_ac
 void ObstacleStopPlannerNode::publishDebugData(
   const PlannerData & planner_data, const double current_acc)
 {
+  mutex_.lock();
   const auto & current_vel = current_velocity_ptr_->twist.twist.linear.x;
```
Comment (yukkysaito) [inline planning/obstacle_stop_planner/src/node.cpp:1537]:
```suggestion
  const auto current_vel = current_velocity_ptr_->twist.twist.linear.x;
```
same comment
https://github.com/autowarefoundation/autoware.universe/pull/1330/files#r919644038

File: planning/obstacle_stop_planner/src/node.cpp (hunk 1 above)
Comment (satoshi-ota) [reply planning/obstacle_stop_planner/src/node.cpp]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/1330/commits/fc819f857be1b2e812d5a9b062e9e8ba1c7b9bd4

File: planning/obstacle_stop_planner/src/node.cpp (hunk 2 above)
Comment (satoshi-ota) [reply planning/obstacle_stop_planner/src/node.cpp]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/1330/commits/fc819f857be1b2e812d5a9b062e9e8ba1c7b9bd4

File: planning/obstacle_stop_planner/src/node.cpp [hunk 3]
```diff
@@ -1518,7 +1533,10 @@ void ObstacleStopPlannerNode::resetExternalVelocityLimit(const double current_ac
 void ObstacleStopPlannerNode::publishDebugData(
   const PlannerData & planner_data, const double current_acc)
 {
-  const auto & current_vel = current_velocity_ptr_->twist.twist.linear.x;
+  mutex_.lock();
+  const auto current_vel = current_velocity_ptr_->twist.twist.linear.x;
+  mutex_.unlock();
+
```
Comment (yukkysaito) [inline planning/obstacle_stop_planner/src/node.cpp:1539]: I think it is better to add current_vel as a argument
```suggestion
void ObstacleStopPlannerNode::publishDebugData(
  const PlannerData & planner_data, const double current_acc, const double current_vel)
  {
```

File: planning/obstacle_stop_planner/src/node.cpp [hunk 4]
```diff
@@ -990,13 +996,17 @@ SlowDownSection ObstacleStopPlannerNode::createSlowDownSection(
...
+  mutex_.lock();
   if (!current_velocity_ptr_) {
     // TODO(Satoshi Ota)
+    mutex_.unlock();
     return SlowDownSection{};
   }
 
+  const auto current_vel = current_velocity_ptr_->twist.twist.linear.x;
+  mutex_.unlock();
+
```
Comment (yukkysaito) [inline planning/obstacle_stop_planner/src/node.cpp:1008]:
```suggestion
  const VehicleInfo & vehicle_info, const double current_acc, const double current_vel)
{
```

File: planning/obstacle_stop_planner/src/node.cpp [hunk 5]
```diff
@@ -1492,7 +1504,10 @@ void ObstacleStopPlannerNode::setExternalVelocityLimit()
 
 void ObstacleStopPlannerNode::resetExternalVelocityLimit(const double current_acc)
 {
+  mutex_.lock();
   const auto current_vel = current_velocity_ptr_->twist.twist.linear.x;
+  mutex_.unlock();
+
```
[the remark this answers is not in the document]
Comment (satoshi-ota) [reply planning/obstacle_stop_planner/src/node.cpp]: Fixed inhttps://github.com/autowarefoundation/autoware.universe/pull/1330/commits/a472dd9417fbf6fc1b17998501f87a44e393ebdb

File: planning/obstacle_stop_planner/src/node.cpp (hunk 3 above)
Comment (satoshi-ota) [reply planning/obstacle_stop_planner/src/node.cpp]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/1330/commits/a472dd9417fbf6fc1b17998501f87a44e393ebdb

File: planning/obstacle_stop_planner/src/node.cpp (hunk 4 above)
Comment (satoshi-ota) [reply planning/obstacle_stop_planner/src/node.cpp]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/1330/commits/a472dd9417fbf6fc1b17998501f87a44e393ebdb

**Comment by satoshi-ota** [comment]
@taikitanaka3 Fixed in https://github.com/autowarefoundation/autoware.universe/pull/1330/commits/9eb1466593f2c46e8b8fbee4d729b85957198751

**Comment by taikitanaka3** [comment]
@satoshi-ota 
[Q]
Is there any way to reproduce this bug? and to prove this is fixed?

**Comment by satoshi-ota** [comment]
> @satoshi-ota [Q] Is there any way to reproduce this bug? and to prove this is fixed?

hmm... I don't know that whether the reproduce conditions exist or not. @s-azumi Do you know about that condition ?
Anyway, lock missing may cause critical bug, so I fixed it.

**Comment by s-azumi** [comment]
@satoshi-ota 
The conditions for reproducing this problem are unknown.
I tried to reproduce the problem with launching logging_simulator and playing rosbag, but this problem did not reappear.

**Review by taikitanaka3 — APPROVED** [review]
LGTM

**Pull request merged**